zipline/testing/core.py

```diff
@@ -465,31 +327,25 @@ def make_trade_data_for_asset_info(dates,
     volumes = (volume_sid_deltas + volume_date_deltas[:, None]) + volume_start
 
     for j, sid in enumerate(sids):
-        start_date = asset_info[sid]['start_date']
-        end_date = asset_info[sid]['end_date']
+        start_date = asset_info.loc[sid, 'start_date']
+        end_date = asset_info.loc[sid, 'end_date']
         # Normalize here so the we still generate non-NaN values on the minutes
         # for an asset's last trading day.
         for i, date in enumerate(dates.normalize()):
             if not (start_date <= date <= end_date):
-                prices[i, j] = np.nan
+                prices[i, j] = 0
                 volumes[i, j] = 0
 
-        if frequency == 'daily':
-            dates_arr = [date.value for date in dates]
-        else:
-            dates_arr = dates
-
-        df = pd.DataFrame({
-            "open": prices[:, j],
-            "high": prices[:, j],
-            "low": prices[:, j],
-            "close": prices[:, j],
-            "volume": volumes[:, j],
-            date_field: dates_arr
-        })
-
-        if frequency == 'minute':
```

Comment:
no, I changed this to use the index instead of the name `day` to make them the same